src/app/core/services/schedule/schedule-generator.service.ts

```diff
@@ -125,15 +135,15 @@ export class ScheduleGeneratorService {
     assessment: Assessment,
     indexOffset: number,
     refTimestamp,
-    type: AssessmentType
+    type: AssessmentType,
+    scheduleYearCoverage: number
   ): Task[] {
     const { repeatP, repeatQ } = this.getRepeatProtocol(
       assessment.protocol,
       type
     )
     let iterTime = refTimestamp
-    const endTime =
-      iterTime + getMilliseconds({ years: DefaultScheduleYearCoverage })
+    const endTime = iterTime + getMilliseconds({ years: scheduleYearCoverage })
```

Comment:
Can't we just call the `getScheduleYearCoverage()` here instead of passing down the  `scheduleYearCoverage` to ensure minimal changes?

Reply:
I've made the change to place the getting of the schedule year coverage in `buildTaskForSingleAssessment()`.
So we fetch it first and store it in the `SCHEDULE_YEAR_COVERAGE` variable, then get it inside `buildTaskforSingleAssessment()`. Is this fine? I didn't directly put the fetching from remote config inside because this will then return promises, and will require changes in more parts of the code as well. 😅